dbcrossbarlib/src/clouds/gcloud/auth.rs

```diff
@@ -0,0 +1,98 @@
+//! Authentication support for Google Cloud.
+
+use dirs;
+use hyper::{self, client::connect::HttpConnector};
+use hyper_rustls::HttpsConnector;
+use serde_json;
+use std::{env, path::PathBuf};
+pub(crate) use yup_oauth2::AccessToken;
+use yup_oauth2::{
+    ApplicationSecret, ConsoleApplicationSecret, InstalledFlowReturnMethod,
+    ServiceAccountKey,
+};
+
+use crate::common::*;
+
+/// The connector type used to create `hyper` connections.
+pub(crate) type HyperConnector = HttpsConnector<HttpConnector>;
+
+pub(crate) type Authenticator =
+    yup_oauth2::authenticator::Authenticator<HyperConnector>;
+
+/// The path to the file where we store our OAuth2 tokens.
+fn token_file_path() -> Result<PathBuf> {
+    Ok(dirs::data_local_dir()
+        .ok_or_else(|| {
+            format_err!("cannot find directory to store authentication keys")
+        })?
+        .join("dbcrossbar-gcloud"))
+}
+
+/// Get the service account key needed to connect a server app to BigQuery.
+fn service_account_key() -> Result<ServiceAccountKey> {
+    let json = env::var("GCLOUD_SERVICE_ACCOUNT_KEY")
+        .context("could not find GCLOUD_SERVICE_ACCOUNT_KEY value")?;
```

Comment:
new requirement?

Reply:
Yeah, `gcloud auth` uses a whole weird custom auth system that has nothing to do with anybody else, that requires boto config files, lots of env vars, etc. We're going to use very simple env-based auth instead.